### PR TITLE
Update session when expired

### PR DESCRIPTION
**Pull request by haiyingweng**
## Overview

We want to update user session whenever the current session expires

## Changes Made

- Added a function to validate session token
- validate session token before each request

## Test Coverage

manually tested that update session works

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

  

</details>

### CODE REVIEW

File: src/utils/requests.ts [hunk 1]
```diff
@@ -32,12 +32,52 @@ const post = async (url: string, body?: any, headers?: any) => {
...
 
-export const setAuthHeader = (token: string | null) => {
-    const header = `Bearer ${token ? token : localStorage.getItem('accessToken')}`
-    axios.defaults.headers.common['Authorization'] = header
+const standardHeaders = () => {
+    const userJson = localStorage.getItem('user')
+    if (userJson == null) {
+        return {}
+    }
+    else {
```
Comment (danielVebman) [inline src/utils/requests.ts:40]: Inline

File: src/utils/requests.ts [hunk 2]
```diff
@@ -32,12 +32,52 @@ const post = async (url: string, body?: any, headers?: any) => {
...
+        const user = JSON.parse(userJson)
```
Comment (danielVebman) [inline src/utils/requests.ts:41]: Can this ever  give null or error? Will accessing a property of null give an error?

File: src/utils/requests.ts [hunk 3]
```diff
@@ -32,12 +32,52 @@ const post = async (url: string, body?: any, headers?: any) => {
...
+        return {
+            headers: { Authorization: `Bearer ${user.sessionAuthorization.updateToken}` }
+        }
+    }
```
Comment (danielVebman) [inline src/utils/requests.ts:58]: Same comments here as in above. The code is very similar? Can you refactor to rescue repetition?

File: src/utils/requests.ts [hunk 4]
```diff
@@ -32,12 +32,52 @@ const post = async (url: string, body?: any, headers?: any) => {
...
 }
 
-export const validateToken = async () => { }
+const validateToken = async () => {
+    const userJson = localStorage.getItem('user')
+    if (userJson != null) {
```
Comment (danielVebman) [inline src/utils/requests.ts:63]: Can it be that userJson == null? What happens then?

**Review by danielVebman — APPROVED** [review]
Neat solution! Looks familiar :)

Just a couple questions

**Review by Npikielny — APPROVED** [review]
Looks good to me!

**Pull request merged**